src/test/java/com/wikia/webdriver/common/core/elemnt/Wait.java

```diff
@@ -395,7 +396,30 @@ public String toString() {
     }
   }
 
-  public void forUrlContains(String text){
+  public void forSuccessfulResponseByUrlPattern(final NetworkTrafficInterceptor trafficInterceptor,
+                                                final String pattern) {
+    changeImplicitWait(0, TimeUnit.SECONDS);
+    try {
+      wait.until(
+          new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver webDriver) {
+              HarEntry entry = trafficInterceptor.getMatchingEntryByUrlPatternFromHar(pattern);
+              return entry != null && entry.getResponse().getStatus() < 400;
+            }
+
+            @Override
+            public String toString() {
+              return "successful response matching pattern: " + pattern;
```

Comment:
This method will be called on error. Could you change the message and show that it's the error message?